Feature/logo page

Dragon88Power opened the pull request.

Dragon88Power committed:
Intro Page

Dragon88Power committed:
missed a line

Dragon88Power committed:
Front Page

Completed front page

Dragon88Power committed:
Changed the width and height of logo and font

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,48 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+//import logo from "../assets/icons/logo.png";
```
delete if not needed

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,48 @@
+import PropTypes from "prop-types";
+//import logo from "../assets/icons/logo.png";
+
+//console.log(logo);
```
delete

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,48 @@
+                <body style={backgroundStyle}>
+                
+                <center>
+                    <br></br><br></br><br></br><br></br><br></br><br></br><br></br>
```
lmao don't use line breaks - even if its centred on your own screen, if you view it on different screen sizes it won't be centred
consider wrapping your icon + text in a **div** and styling it, perhaps with align-items : center or something like that

Dragon88Power replied:
Can't implement align items or an alternative to solve this issue

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,48 @@
+                    <br></br><br></br><br></br><br></br><br></br><br></br><br></br>
+
+                    <img src={"/static/assets/icons/logo.png"} alt = "app logo" width="200" height="100" />
+                    <h1 style = {nameStyle1}> <br></br>
```
don't need the line breaks
if you want space between your elements, read about margins and paddings in styling

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,48 @@
+    render(){
+               
+            const backgroundStyle = {
+                backgroundColor: "#32669d", 
```
capitalise the 'd' for consistency and neatness

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,48 @@
+            };
+            const nameStyle1 = {
+                color: "white", 
+                fontSize:"50px",  
```
sorry pedantic - add space between fontSize:  and "50px",

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,48 @@
+            };
+            const nameStyle2 = {
+                color: "white", 
+                fontSize:"20px",  
```
refer to above

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,48 @@
+                    <h2 style = {nameStyle2}>
+                        ROSTER
+                    </h2>                                      
+                </center>
```
I dont think **center** is actually supported anymore, so refer to what i said above and center the **div** with styling (perhaps look up justify-items, or text-align property should work)

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,48 @@
+               
+            const backgroundStyle = {
+                backgroundColor: "#32669d", 
+                width: "1919.9px", 
```
i don't think you need to specify height and width - otherwise on screens bigger than these itll look like a blue box with a thick white border

glui2 replied:
if it doesn't work without them just use 100% for both

Dragon88Power replied:
Having troubles in filling the height of the page with the background. Similar issue to: https://stackoverflow.com/questions/46769706/css-100-height-not-working-in-react-app/46770041
Could you have a look

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,48 @@
+            };
+
+            return(
+                <body style={backgroundStyle}>
```
use **div** instead of **body**, as i think it will already be surrounded by a body tag

glui2 reviewed (changes requested):
whilst it probably works for your screen, an important part of developing frontend is responsiveness - ie will it look the same on different screen sizes? look up some of the stylings i mentioned, you'll use them a lot

Dragon88Power committed:
minor text fixes, more responsive

Made minor text fixes, design is more responsive. Vertical centering not working

Dragon88Power committed:
Responsive design 

Implemented centering and it is more responsive to design

Dragon88Power committed:
Changed html tag

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,54 @@
+            position: "absolute",
+            top: "25%",
+            left: "20%"
+
```
remove extra line

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,54 @@
+            top: "25%",
+            left: "20%"
+
+        }
```
need semi colon

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,54 @@
+        const imgStyle = {
+            display: "block",
+            margin: "auto"
+        }
```
need semi colon

glui2 replied:
then maybe a new line before the return function

glui2 commented on django-drf-react-quickstart/mcbcapp/frontend/src/components/IntroPage.js:
```diff
@@ -0,0 +1,54 @@
+            margin: "auto"
+        }
+        return(
+            <body style={backgroundStyle}>
```
your whole component will already be nested in a **body** tag so this actually isnt needed

glui2 replied:
use div instead

Dragon88Power committed:
switched body with div to avoid error

glui2 reviewed (changes requested):
nice work, just neaten up the code pls
dont forget semi colons, remove random empty lines but have empty lines if it looks better :)

Dragon88Power committed:
Minor text fixes
